ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -224,363 +301,231 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
-        ) {
-
-      case class Reject(
-          override val cause: String,
-          _err: LfInterpretationError.ContractNotActive,
-      )(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = cause
-          ) {
-        override def resources: Seq[(ErrorResource, String)] = Seq(
-          (ErrorResource.ContractId, _err.coid.coid)
-        )
-      }
-
-    }
-
-    object LookupErrors extends ErrorGroup {
-
-      @Explanation("""This error occurs if the Damle interpreter can not find a referenced contract. This
-          |can be caused by either the contract not being known to the participant, or not being known to
-          |the submitting parties or already being archived.""")
-      @Resolution("This error type occurs if there is contention on a contract.")
-      object ContractNotFound
-          extends ErrorCode(
-            id = "CONTRACT_NOT_FOUND",
-            ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
-          ) {
-
-        case class Reject(
-            override val cause: String,
-            _cid: Value.ContractId,
-        )(implicit
-            loggingContext: ContextualizedErrorLogger
-        ) extends LoggingTransactionErrorImpl(
-              cause = cause
-            ) {
-          override def resources: Seq[(ErrorResource, String)] = Seq(
-            (ErrorResource.ContractId, _cid.coid)
-          )
-        }
-
-      }
-
-      @Explanation(
-        """This error occurs if the Damle interpreter can not resolve a contract key to an active contract. This
-          |can be caused by either the contract key not being known to the participant, or not being known to
-          |the submitting parties or the contract representing the key has already being archived."""
-      )
-      @Resolution("This error type occurs if there is contention on a contract.")
-      object ContractKeyNotFound
-          extends ErrorCode(
-            id = "CONTRACT_KEY_NOT_FOUND",
-            ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
-          ) {
-
-        case class Reject(
-            override val cause: String,
-            _key: GlobalKey,
-        )(implicit
-            loggingContext: ContextualizedErrorLogger
-        ) extends LoggingTransactionErrorImpl(
-              cause = cause
-            ) {
-          override def resources: Seq[(ErrorResource, String)] = Seq(
-            (ErrorResource.ContractKey, _key.toString())
+    @Resolution("Inspect the reason given and correct your application.")
+    object InvalidArgument
+        extends ErrorCode(id = "INVALID_ARGUMENT", ErrorCategory.InvalidIndependentOfSystemState) {
+      case class Reject(_reason: String)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = s"The submitted command has invalid arguments: ${_reason}"
```

Comment:
Nit-picking:
Explanation mentions singular: "(...) an invalid argument."
but here we have plural "(..) invalid arguments: (...)"

Reply:
This error accepts very generic reasons. Not ideal, but I think plural accommodates a wider set of reasons better.